che-editors.yaml

```diff
@@ -1,373 +1,456 @@
-    icon: https://raw.githubusercontent.com/theia-ide/theia/master/logo/theia-logo-no-text-black.svg?sanitize=true
-    repository: https://github.com/eclipse-che/che-theia
-    firstPublicationDate: "2019-03-07"
-    endpoints:
-      - name: "theia"
-        public: true
-        targetPort: 3100
-        attributes:
-          protocol: http
-          type: ide
-          secure: true
-          cookiesAuthEnabled: true
-          discoverable: false
-      - name: "webviews"
-        public: true
-        targetPort: 3100
+        container:
+          image: 'quay.io/eclipse/che-theia:next'
+          env:
+            - name: THEIA_PLUGINS
+              value: 'local-dir:///plugins'
+            - name: HOSTED_PLUGIN_HOSTNAME
+              value: 0.0.0.0
+            - name: HOSTED_PLUGIN_PORT
+              value: '3130'
+            - name: THEIA_HOST
+              value: 127.0.0.1
+          volumeMounts:
+            - name: plugins
+              path: /plugins
+            - name: theia-local
+              path: /home/theia/.theia
+          mountSources: true
+          memoryLimit: 512M
+          cpuLimit: 1500m
+          cpuRequest: 100m
+          endpoints:
+            - name: theia
+              exposure: public
+              secure: true
+              targetPort: 3100
+              attributes:
+                type: ide
+                cookiesAuthEnabled: true
+                discoverable: false
+              protocol: http
+            - name: webviews
+              targetPort: 3100
+              exposure: public
+              secure: true
+              protocol: http
+              attributes:
+                type: webview
+                cookiesAuthEnabled: true
+                discoverable: false
+                unique: true
+            - name: mini-browser
+              targetPort: 3100
+              exposure: public
+              secure: true
+              protocol: http
+              attributes:
+                type: mini-browser
+                cookiesAuthEnabled: true
+                discoverable: false
+                unique: true
+            - name: theia-dev
+              targetPort: 3130
+              exposure: public
+              protocol: http
+              attributes:
+                type: ide-dev
+                discoverable: false
+            - name: theia-redirect-1
+              targetPort: 13131
+              exposure: public
+              protocol: http
+              attributes:
+                discoverable: false
+            - name: theia-redirect-2
+              targetPort: 13132
+              exposure: public
+              protocol: http
+              attributes:
+                discoverable: false
+            - name: theia-redirect-3
+              targetPort: 13133
+              exposure: public
+              protocol: http
+              attributes:
+                discoverable: false
+            - name: terminal
```

Comment:
this endpoint should be moved to the che-machine-exec container

Reply:
done